icrawler/downloader.py

```diff
@@ -5,6 +5,7 @@
 import threading
 
 import requests
+from urllib.parse import urlparse
```

Comment:
To make the code compatible with both python 2 and 3, the package `six` should be used instead of directly import `urllib`, so this line can be modified as
```python
from six.moves.urllib.parse import urlparse
```